pkg/build/api/v1/types.go

```diff
@@ -52,8 +52,75 @@ type BuildSpec struct {
+type BuildTriggerInfo struct {
+	// Details is a placeholder for  information about who triggered a build if
+	// it's either a user, a token or something else
+	Details BuildTriggerDetails `json:"details,omitempty" description:"generic information about a trigerred build and/or about the user that triggered it"`
+
+	// CIJobDetails is a generic placeholder that should store information about
+	// jobs from CI plugins that integrate with openshift and triggered a
+	// specific build
+	CIDetails CIJobDetails `json:"ciDetails,omitempty" description:"CI job information that triggered a build"`
+
+	// GitWebHookInfo holds information about a build that was triggered by a
+	// gitwebhook including its refs, branch, last commit id, etc. It also stores
+	// the type of plugin that triggered it, if it was a generic one, github,
+	// bitbucket, etc.
+	GitWebHook GitWebHookInfo `json:"gitWebHook,omitempty" description:"general information about a webhook that triggered this build"`
+
+	// ImageChangeBuildInfo - Stores info about the image that triggered a new
+	// build
+	ImageChange ImageChangeBuildInfo `json:"imageChange,omitempty" description:"information about an image change build trigger"`
+
+	// ConfigChangeBuildInfo - stores info about the last version of the config
+	// that triggered this build
+	ConfigChange ConfigChangeBuildInfo `json:"configChange,omitempty" description:"generic information about a configchange that triggered a build"`
+}
+
+// BuildTriggerDetails holds information about who/what triggered a build
+type BuildTriggerDetails string
+
+// CIJobDetails holds information about the CI job that triggered a
+// build
+type CIJobDetails struct {
+	// Type is type of the CI that started this. "jenkins", "travis", etc.
+	Type string `json:"type,omitempty" description:"type of CI that triggered the build"`
+
+	// JobURL holds the url that points to the job that triggered a specific
+	// build
+	JobURL string `json:"jobURL,omitempty"  description:"the URL with information about the specific job tha triggered a build"`
+}
+
+// GitWebHookInfo holds information about a git webhook that triggered a build
+type GitWebHookInfo struct {
+	// GitTriggerType is the git webhook trigger type (github, generic, etc)
+	GitTriggerType BuildTriggerType `json:"gitTriggerType,omitempty"`
+	// Git is where the information about the current git webhook is stored
+	GitDetails *GitInfo `json:"gitDetails,omitempty"`
+}
+
+// ImageChangeBuildInfo holds information about the image that triggered a new
+// build
+type ImageChangeBuildInfo struct {
+	// PreviousImageID is the old image ID before the image change trigger fires
+	PreviousImageID string `json:"previousImageID,omitempty" description:"the image ID before the image change trigger fired"`
+	// CurrentImageID is the image ID of the image that triggered a build
+	CurrentImageID string `json:"currentImageID,omitempty" description:"the image ID that triggered the build"`
```

Comment:
My hold was partially because I may have changes to the design (because we
need triggers across objects besides builds and deployments) and I don't
want you to potentially waste time if we change direction fundamentally.

On Thu, Mar 10, 2016 at 11:15 AM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/7518#discussion_r55703873>:
>
> > +
> > +// GitWebHookInfo holds information about a git webhook that triggered a build
> > +type GitWebHookInfo struct {
> > +	// GitTriggerType is the git webhook trigger type (github, generic, etc)
> > +	GitTriggerType BuildTriggerType `json:"gitTriggerType,omitempty"`
> > +	// Git is where the information about the current git webhook is stored
> > +	GitDetails *GitInfo `json:"gitDetails,omitempty"`
> > +}
> > +
> > +// ImageChangeBuildInfo holds information about the image that triggered a new
> > +// build
> > +type ImageChangeBuildInfo struct {
> > +	// PreviousImageID is the old image ID before the image change trigger fires
> > +	PreviousImageID string `json:"previousImageID,omitempty" description:"the image ID before the image change trigger fired"`
> > +	// CurrentImageID is the image ID of the image that triggered a build
> > +	CurrentImageID string `json:"currentImageID,omitempty" description:"the image ID that triggered the build"`
>
> similar comments here :)
>
> (though to be fair, i'm not sure we care about previous image, either..
> just the current image (the one that triggered this build) should be of
> interest.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7518/files#r55703873>.
>